mcp/src/main/java/io/airlift/mcp/reference/ReferenceServerTransport.java

```diff
@@ -0,0 +1,189 @@
+import io.modelcontextprotocol.spec.McpSchema.JSONRPCNotification;
+import io.modelcontextprotocol.spec.McpSchema.JSONRPCRequest;
+import io.modelcontextprotocol.spec.McpSchema.JSONRPCResponse;
+import io.modelcontextprotocol.spec.McpSchema.JSONRPCResponse.JSONRPCError;
+import io.modelcontextprotocol.spec.McpStatelessServerTransport;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import reactor.core.publisher.Mono;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.stream.Collectors;
+
+import static io.airlift.mcp.McpMetadata.CONTEXT_REQUEST_KEY;
+import static io.modelcontextprotocol.common.McpTransportContext.KEY;
+import static io.modelcontextprotocol.spec.McpSchema.ErrorCodes.INTERNAL_ERROR;
+import static io.modelcontextprotocol.spec.McpSchema.ErrorCodes.INVALID_REQUEST;
+import static io.modelcontextprotocol.spec.McpSchema.deserializeJsonRpcMessage;
+import static jakarta.servlet.http.HttpServletResponse.SC_ACCEPTED;
+import static jakarta.servlet.http.HttpServletResponse.SC_BAD_REQUEST;
+import static jakarta.servlet.http.HttpServletResponse.SC_INTERNAL_SERVER_ERROR;
+import static jakarta.servlet.http.HttpServletResponse.SC_METHOD_NOT_ALLOWED;
+import static jakarta.servlet.http.HttpServletResponse.SC_NOT_FOUND;
+import static jakarta.servlet.http.HttpServletResponse.SC_OK;
+import static jakarta.servlet.http.HttpServletResponse.SC_SERVICE_UNAVAILABLE;
+import static jakarta.ws.rs.core.HttpHeaders.ACCEPT;
+import static jakarta.ws.rs.core.MediaType.APPLICATION_JSON;
+import static jakarta.ws.rs.core.MediaType.SERVER_SENT_EVENTS;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Objects.requireNonNull;
+
+// based off of io.modelcontextprotocol.server.transport.HttpServletStatelessServerTransport.java
+public class ReferenceServerTransport
+        extends HttpServlet
+        implements McpStatelessServerTransport
+{
+    private static final Logger logger = Logger.get(ReferenceServerTransport.class);
+
+    private final McpJsonMapper mcpJsonMapper;
+    private final String mcpEndpoint;
+    private volatile McpStatelessServerHandler mcpHandler;
+    private volatile boolean isClosing;
+
+    @Inject
+    public ReferenceServerTransport(McpMetadata metadata, McpJsonMapper mcpJsonMapper)
+    {
+        this.mcpJsonMapper = requireNonNull(mcpJsonMapper, "jsonMapper is null");
+        this.mcpEndpoint = metadata.uriPath();
+    }
+
+    @Override
+    public void setMcpHandler(McpStatelessServerHandler mcpHandler)
+    {
+        this.mcpHandler = requireNonNull(mcpHandler, "mcpHandler is null");
+    }
+
+    @Override
+    public Mono<Void> closeGracefully()
+    {
+        return Mono.fromRunnable(() -> isClosing = true);
+    }
+
+    @Override
+    public void destroy()
+    {
+        closeGracefully().block();
```

Comment:
just `isClosing = true` ? Do we need Mono here? :)

Reply:
We need to call `closeGracefully()` and it returns a `Mono` immediately. So, yeah. My vote is stop using the reference SDK altogether. It's not giving us much. I posted a straw man implementation that removes it.